document/core/syntax/modules.rst

```diff
@@ -21,6 +22,7 @@ and provide initialization in the form of :ref:`active <syntax-active>` and :ref
      \MGLOBALS~\vec(\global), \\&&&&
      \MELEM~\vec(\elem), \\&&&&
      \MDATA~\vec(\data), \\&&&&
+     \MDATACOUNT~\u32^?, \\&&&&
```

Comment:
Hm, why do you need to extend the abstract syntax (and validation)? The side condition is already handled as a detail of the binary format, so can't the rest remain as before? IIRC, there was no corresponding change in the interpreter.

Reply:
We also need to fail when `memory.init` or `data.drop` are used, but the data count section is not present. I forgot that we handled this during decoding -- I'll update it tomorrow.

Reply:
Done.